File: src/main/java/kr/devport/api/domain/wiki/service/WikiAdminProjectQueryService.java
package kr.devport.api.domain.wiki.service;

import kr.devport.api.domain.port.entity.Project;
import kr.devport.api.domain.port.repository.ProjectRepository;
import kr.devport.api.domain.wiki.dto.response.WikiAdminProjectSummaryResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class WikiAdminProjectQueryService {

    private final ProjectRepository projectRepository;

    public List<WikiAdminProjectSummaryResponse> listProjects() {
        return projectRepository.findAllForWikiAdmin().stream()
                .map(this::toSummary)
                .toList();
    }

    private WikiAdminProjectSummaryResponse toSummary(Project project) {
        return WikiAdminProjectSummaryResponse.builder()
                .projectId(project.getId())
                .projectExternalId(project.getExternalId())
                .fullName(project.getFullName())
                .stars(project.getStars() == null ? 0 : project.getStars())
                .language(project.getLanguage() == null || project.getLanguage().isBlank() ? "Unknown" : project.getLanguage())
                .build();
    }
}

File: wiki/service/src/main/kotlin/kr/devport/api/domain/wiki/service/WikiAdminProjectQueryService.kt
package kr.devport.api.domain.wiki.service

import kr.devport.api.domain.port.ProjectView
import kr.devport.api.domain.port.infrastructure.ProjectDirectory
import kr.devport.api.domain.wiki.dto.response.WikiAdminProjectSummaryResponse
import org.springframework.stereotype.Service

@Service
class WikiAdminProjectQueryService(
    private val projectDirectory: ProjectDirectory,
) {
    fun listProjects(): List<WikiAdminProjectSummaryResponse> = projectDirectory.listForWikiAdmin().map { it.toSummary() }

    private fun ProjectView.toSummary(): WikiAdminProjectSummaryResponse =
        WikiAdminProjectSummaryResponse(
            projectId = id,
            projectExternalId = externalId,
            fullName = fullName,
            stars = stars ?: 0,
            language = language?.takeUnless { it.isBlank() } ?: "Unknown",
        )
}
